Sources/SwiftBuildSupport/PackagePIFProjectBuilder+Products.swift

```diff
@@ -995,6 +1001,102 @@ extension PackagePIFProjectBuilder {
         )
         self.builtModulesAndProducts.append(pluginProductMetadata)
     }
+
+    // MARK: - Test Runners
+    mutating func makeTestRunnerProduct(for unitTestProduct: PackagePIFBuilder.ModuleOrProduct) throws {
+        // Only generate a test runner for root packages with tests.
+        guard pifBuilder.delegate.isRootPackage else {
+            return
+        }
+
+        guard let unitTestModuleName = unitTestProduct.moduleName else {
+            fatalError()//throw StringError("")
+        }
+
+        let name = "\(unitTestProduct.name)-test-runner"
+        let moduleName = "\(unitTestModuleName)_test_runner"
+        let guid = PackagePIFBuilder.targetGUID(forModuleName: moduleName)
+
+        let testRunnerTargetKeyPath = try self.project.addTarget { _ in
+            ProjectModel.Target (
+                id: guid,
+                productType: .swiftpmTestRunner,
+                name: name,
+                productName: name
+            )
+        }
+
+        var settings: BuildSettings = self.package.underlying.packageBaseBuildSettings
+        let impartedSettings = BuildSettings()
+
+        settings[.TARGET_NAME] = name
+        settings[.PACKAGE_RESOURCE_TARGET_KIND] = "regular"
+        settings[.PRODUCT_NAME] = "$(TARGET_NAME)"
+        settings[.PRODUCT_MODULE_NAME] = moduleName
+        settings[.PRODUCT_BUNDLE_IDENTIFIER] = "\(self.package.identity).\(name)"
+            .spm_mangledToBundleIdentifier()
+        settings[.EXECUTABLE_NAME] = name
```

Comment:
The PIFBuilder is overriding EXECUTABLE_NAME for a bunch of target types, which sometimes strips extensions and prefixes of product types, we might need to rework some of that